de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingReferencePointNegotiation.java

```diff
@@ -386,30 +395,30 @@ public synchronized boolean localCancel(String errorMsg,
     }
 
     /**
-     * Synchronize the project structures, deleting files and folders that are
-     * not present on the remote side and creating empty folders that do not
-     * exists and the local side.
+     * Synchronize the reference point structures, deleting files and folders
+     * that are not present on the remote side and creating empty folders that
+     * do not exists and the local side.
      * 
-     * @param localProjectMapping
+     * @param localReferencePointMapping
      * @param diffs
-     * @return list of file lists (each for every project) containing the
-     *         missing files that are not present on the local side.
+     * @return list of file lists (each for every reference point) containing
+     *         the missing files that are not present on the local side.
      * @throws IOException
      */
-    protected List<FileList> synchronizeProjectStructures(
-        final Map<String, IProject> localProjectMapping,
+    protected List<FileList> synchronizeReferencePointStructures(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 21 to the 15 allowed. ([squid:S3776](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS3776))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=273&issues=AWaWrfcIsSPV6ZXI_B9I&open=AWaWrfcIsSPV6ZXI_B9I)

[ISSUE_KEY]: AWaWrfcIsSPV6ZXI_B9I